python/pyarrow/types.pxi

```diff
@@ -962,12 +962,30 @@ def tzinfo_to_string(tz):
       name : string
         Time zone name
     """
-    if tz.zone is None:
-        sign = '+' if tz._minutes >= 0 else '-'
-        hours, minutes = divmod(abs(tz._minutes), 60)
+    import pytz
+    import datetime
+
+    def fixed_offset_to_string(offset):
+        seconds = int(offset.utcoffset(None).total_seconds())
+        sign = '+' if seconds >= 0 else '-'
+        minutes, seconds = divmod(abs(seconds), 60)
+        hours, minutes = divmod(minutes, 60)
+        if seconds > 0:
+            raise ValueError('Offset must represent whole number of minutes')
         return '{}{:02d}:{:02d}'.format(sign, hours, minutes)
-    else:
+
+    if isinstance(tz, pytz.tzinfo.BaseTzInfo):
```

Comment:
Apparently, `pytz.UTC` is only a subclass of BaseTzInfo starting from pytz 2018.4 (https://launchpad.net/pytz/+announcement/14962), so with older versions of pytz there are now failures in the tests due to:

```
In [4]: pyarrow.lib.tzinfo_to_string(pytz.utc)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-4-8559e74fba95> in <module>()
----> 1 pyarrow.lib.tzinfo_to_string(pytz.utc)

~/scipy/repos/arrow/python/pyarrow/types.pxi in pyarrow.lib.tzinfo_to_string()

ValueError: Unable to convert timezone `UTC` to string
```

Is there a minimum version of pytz that is officially supported?